transit_odp/organisation/csv/service_codes.py

```diff
@@ -1,59 +1,229 @@
+            accessor=lambda otc_service: otc_service.get(
+                "effective_stale_date_otc_effective_date"
+            ),
+        ),
+    ]
 
     def __init__(self, organisation_id: int):
         self._organisation_id = organisation_id
-        self.org_service_code_to_dataset = list(
-            TXCFileAttributes.objects.filter(
-                revision__dataset__organisation_id=organisation_id
-            )
-            .get_active_live_revisions()
-            .select_related("revision__dataset")
-            .values_list("service_code", "revision__dataset_id")
+        self._object_list = []
+
+    def _update_data(
+        self,
+        service: OTCService,
+        file_attribute: Optional[TXCFileAttributes],
+        seasonal_service: Optional[SeasonalService],
+        staleness_status: Optional[str],
+        require_attention: str,
+    ) -> None:
+        self._object_list.append(
+            {
+                "require_attention": require_attention,
+                "licence_number": service.otc_licence_number,
+                "service_code": service.registration_number,
+                "line_number": service.service_number,
+                "dataset_id": file_attribute and file_attribute.revision.dataset_id,
+                "seasonal_status": seasonal_service
+                and seasonal_service.seasonal_status,
+                "seasonal_start": seasonal_service and seasonal_service.start,
+                "seasonal_end": seasonal_service and seasonal_service.end,
+                "staleness_status": staleness_status,
+                "effective_seasonal_start_date": seasonal_service
+                and seasonal_service.start - timedelta(days=42),
+                "effective_stale_date_end_date": file_attribute
+                and file_attribute.effective_stale_date_end_date,
+                "effective_stale_date_last_modified_date": file_attribute
+                and file_attribute.effective_stale_date_last_modified_date,
+                "last_modified_lt_effective_stale_date_otc": (
+                    service.last_modified.date()
+                    < service.effective_stale_date_otc_effective_date
+                ),
+                "effective_stale_date_otc_effective_date": (
+                    service.effective_stale_date_otc_effective_date
+                ),
+            }
         )
 
-    def get_queryset(self):
-        return OTCService.objects.get_all_without_exempted_ones(self._organisation_id)
+    def get_otc_service_data(self, organisation_id: int) -> None:
+        """
+        Compares an organisation's OTC Services dictionaries list with
+        TXCFileAttributes dictionaries list and the SeasonalService list
+        to determine which OTC Services require attention and which
+        doesn't ie. not live in BODS at all, or live but meeting new
+        Staleness conditions.
+        """
+        otc_map = get_otc_map(organisation_id)
```

Comment:
This function in `transit_odp/publish/requires_attention.py` calls `OTCService.objects.get_otc_data_for_organisation(org_id)` which excludes Exempted and Out of Season services from what's returned, so not sure how using this can give you back all the data you need with those already excluded?